### PR TITLE
fix: add null protection for locations field

### LINKED ISSUE #799: Searching `motionbuscard` causes a blank screen and breaks further searches
### Describe the bug

After searching for `motionbuscard` in the Mobility Database web interface, the application displays a blank screen. Subsequent searches return the same blank screen, indicating an unresolved error.

### Steps/Code to Reproduce

1. Open [mobilitydatabase.org](http://mobilitydatabase.org).
2. Search for `motionbuscard`.
3. Return to the home page.
4. Attempt to search for anything else.

### Expected Results

The search should yield results or indicate "no results" if none are found, with the ability to perform further searches.

### Actual Results

A white (blank) page appears after the initial search for "motionbuscard," persisting for any subsequent search attempts. Errors are displayed in the console.

### Additional notes

I believe that mdb-1920 (having `motionbuscard` provider) is causing issues because a lot of its fields are set as unknown in the catalog repo

### PR DESCRIPTION
**Pull request by davidgamez**
**Summary:**

Fix  #799

**Expected behavior:** 

Having null values in locations don't crash the app.

**Testing tips:**

Follow steps in #799

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Run the unit tests with `./scripts/api-tests.sh` to make sure you didn't break anything
- [ ] Add or update any needed documentation to the repo
- [ ] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [ ] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

**Comment by davidgamez** [comment]
Screenshot 2024-11-06 at 1 49 22 PM

**Review by cka-y — APPROVED** [review]

**Comment by jcpitre** [comment]
@davidgamez I think we need to go around and correct other cases where this happens. e.g. : https://github.com/MobilityData/mobility-feed-api/blob/73ea4bc5c5ace81455161856fcb17c9b3760239e/web-app/src/app/screens/Feed/FeedSummary.tsx#L240
We get an NPE and a white page when trying to display the summary for tld-933

File: web-app/src/app/services/feeds/utils.ts [hunk 1]
```diff
@@ -50,9 +50,9 @@ export type AllDatasetType =
   | undefined;
 
 export function getLocationName(
-  locations: EntityLocations | undefined,
+  locations: EntityLocations | undefined | null,
 ): string {
-  if (locations === undefined) {
+  if (locations === undefined || locations === null) {
```
Comment (jcpitre) [inline web-app/src/app/services/feeds/utils.ts:55]: Can't we use `locations == null` to cover both cases? Why do we use a strict equality operator?

**Comment by isabelle-dr** [comment]
Also a blank page for https://mobilitydatabase.org/feeds/tld-823

**Comment by isabelle-dr** [comment]
Other one https://mobilitydatabase.org/feeds/tld-947

**Comment by emmambd** [comment]
@isabelle-dr It's because you're logged in as a mobilitydata user. It's not a user facing issue - those are unpublished feeds. I can give you an overview of how this new workflow operates on Friday.

**Pull request merged**